### PR TITLE
Create a database based on a diagram

### PR DESCRIPTION
**Pull request by mrkamin**
What does this PR do?
In this PR we create a Database for a clinic based on diagram ...

Description of the task to be completed 👎

we made the clinic Database based on below database.

clinic_diagram

we Create a file named schema_based_on_diagram.sql where we implement the database from the diagram. please check it through below linke (https://github.com/mrkamin/Vet-Clinic/blob/Diagram-branch/schema_based_on__diagram.sql).

### CODE REVIEW

File: schema_based_on__diagram.sql [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    CONSTRAINT fk_treatment FOREIGN KEY(treatment_id) REFERENCES treatments(id) ON DELETE CASCADE,
+    CONSTRAINT fk_invoice FOREIGN KEY(invoice_id) REFERENCES invoices(id) ON DELETE CASCADE
+);
+
+/* Creates medical_treatment_histories table*/
+CREATE TABLE medical_treatment_histories (
+    medical_id INT NOT NULL REFERENCES medical_histories(id),
+    treatment_id INT NOT NULL REFERENCES treatments(id),
+    PRIMARY KEY (medical_id, treatment_id) 
+);
```
Comment (franclobo) [inline schema_based_on__diagram.sql]: - [x] Well done creating all the tables like in the ERD, please 🙏🏽 ensure to create indexes for all foreign keys, there should be six of them 👇🏽 

`CREATE INDEX ON medical_histories (patients_id);`

**Review by franclobo — COMMENTED** [review]
Hi @mrkamin, @miltonHenschel, and @Iqbal-Elham , 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- Correct Git flow ✔️ 
- Descriptive PR ✔️ 
- Correct schema ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on__diagram.sql (hunk 1 above)
Comment (mrkamin) [reply schema_based_on__diagram.sql]: @franclobo,
thanks for your good and valuable feedback,
I am not sure about the suggestion you have given, I kindly request you if you take more details. I think it is not required.

Comment (mrkamin) [reply schema_based_on__diagram.sql]: I have created Index as you suggest but you mention we have to have six but I could create five, no Idea about the sixth one could you suggest me a way of creating it, 
Create Indexes

File: schema_based_on__diagram.sql [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+CREATE INDEX ON medical_histories (patient_id);
+
+CREATE INDEX ON invoice_items (invoice_id);
+
+CREATE INDEX ON invoice_items (treatment_id);
+
+CREATE INDEX ON invoices (medical_history_id);
+
+CREATE INDEX ON medical_treatment_histories (treatment_id);
```
Comment (gemmen29) [inline schema_based_on__diagram.sql]: - [x] Good job on implementing almost all foreign key indexes as requested by the previous, But there is still only one foreign key that didn't have an index. Kindly add an index for the foreign key `medical_id` inside `medical_treatment_histories` as per this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md#:~:text=Remember%20to%20add%20the%20FK%20indexes.).

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Highlights💯 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on__diagram.sql (hunk 2 above)
Comment (mrkamin) [reply schema_based_on__diagram.sql]: @gemmen29 
thanks for your good and valuable feedback,
added
Capture

**Review by tugsem — APPROVED** [review]
## STATUS: APPROVED 👍🎯  
Hello @mrkamin👋 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⭐ 
✔️ correct use of gitflow
✔️ linters are passing
✔️ professional documentation
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
@tugsem,
thanks for your good and valuable feedback.

**Pull request merged**